tasks/service.py

```
@@ -912,39 +975,83 @@ def _build_spec(self, task: Task, agent: AgentDefinition | None) -> TaskSpec:
+                # Some store impls are async — try the async path.
+                import asyncio as _asyncio
+                try:
+                    loop = _asyncio.get_event_loop()
+                    if loop.is_running():
+                        # We're inside an event loop already; can't call run
+                        # until. Fall through to None — the runtime will run
+                        # against the default workspace.
+                        company = None
+                    else:
+                        company = _asyncio.run(store.get_company(company_id))
+                except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)